### PR TITLE
Dl staging

### LINKED ISSUE #2: Expose edge insets for layout
- Expose the edge insets to allow for layout customization

### LINKED ISSUE #3: Block completion handlers

### LINKED ISSUE #4: Set button image with size
Add functionality to specify button size with an image. 

Default layout behavior will be to lay the buttons out along the bottom of the button area (baseline)

### PR DESCRIPTION
**Pull request by Dfowj**
Closes #4 - Custom button sizes
Closes #3 - Implement completion handler blocks
Closes #2 - Exposes edge insets

Adds demo cases for variable button size layouts

### CODE REVIEW

File: TTAlertView/TTAlertView.h [hunk 1]
```diff
@@ -8,11 +8,15 @@
 
 #import <UIKit/UIKit.h>
 
+typedef void (^TTAlertViewCompletionHandler)(NSInteger index);
+
 @protocol TTAlertViewDelegate;
 
 @interface TTAlertView : UIView
+{}
```
Comment (danielrhammond) [inline TTAlertView/TTAlertView.h:16]: remove

File: TTAlertView/TTAlertView.h [hunk 2]
```diff
@@ -21,13 +25,10 @@
...
-- (id)initWithTitle:(NSString *)title message:(NSString *)message delegate:(id)delegate cancelButtonTitle:(NSString *)cancelButtonTitle otherButtonTitles:(NSString *)otherButtonTitles, ...;
-- (void)show;
-- (void)dismissWithClickedButtonIndex:(NSInteger)index animated:(BOOL)animated;
-- (void)addButtonWithTitle:(NSString *)title;
-- (NSString *)buttonTitleAtIndex:(NSInteger)index;
+#pragma mark - TTAlertView properties
 
-// TTAlertView properties and methods exposed for customization
+@property (nonatomic, copy) TTAlertViewCompletionHandler cancelButtonCompletionHandler;
+@property (nonatomic, copy) TTAlertViewCompletionHandler otherButtonsCompletionHandler;
```
Comment (danielrhammond) [inline TTAlertView/TTAlertView.h]: per our discussion remove cancelButtonCompletionHandler, rename other to buttonAction

File: TTAlertView/TTAlertView.h (hunk 1 above)
Comment (Dfowj) [inline TTAlertView/TTAlertView.h:16]: for some reason xcode won't annotate the following #pragma mark - unless i add the brackets.

Comment (kconner) [inline TTAlertView/TTAlertView.h:16]: I have the same problem right after @implementation. No idea if it's even possible to correct it. But personally I'd err on the side of the code itself looking readable.

File: TTAlertView/TTAlertView.m [hunk 3]
```diff
@@ -377,12 +507,27 @@ - (void)setupView
 
 - (void)cancelButtonAction:(id)sender
 {
+    if(self.buttonActionHandler != nil) {
```
Comment (danielrhammond) [inline TTAlertView/TTAlertView.m]: `if(self.buttonActionHandler) {`

File: TTAlertView/TTAlertView.m [hunk 4]
```diff
@@ -377,12 +507,27 @@ - (void)setupView
...
+        }
+    }
     [self dismissWithClickedButtonIndex:self.cancelButtonIndex animated:YES];
 }
 
 - (void)otherButtonAction:(id)sender
 {
-    [self dismissWithClickedButtonIndex:[self.buttons indexOfObject:sender] + self.firstOtherButtonIndex animated:YES];
+    NSInteger index = [self.buttons indexOfObject:sender];
+    if(self.buttonActionHandler != nil) {
```
Comment (danielrhammond) [inline TTAlertView/TTAlertView.m]: `if(self.buttonActionHandler) {`

**Comment by danielrhammond** [comment]
After fixing those two style comments it looks ready to merge to me

**Pull request merged**